mlir/lib/Dialect/Tosa/IR/TosaCanonicalizations.cpp

```diff
@@ -374,6 +374,19 @@ void ClampOp::getCanonicalizationPatterns(RewritePatternSet &results,
   results.add<ClampClampOptimization>(context);
 }
 
+OpFoldResult ClampOp::fold(FoldAdaptor adaptor) {
+  // TODO: This can generalize to any cast (or other operation)
+  // where the output values are within a computable range.
+  if (auto cast = llvm::dyn_cast_or_null<CastOp>(getInput().getDefiningOp())) {
```

Comment:
Add a comment on why this fold makes sense, something about the range being wider than the type width we just casted from.